packages/destination-actions/src/lib/test-data.ts

```diff
@@ -0,0 +1,75 @@
+import { ActionDefinition, DestinationDefinition } from '@segment/actions-core'
+
+const testData: { [key: string]: any } = {
+  boolean: true,
+  datetime: '2021-01-17',
+  integer: 1,
+  number: 5,
+  string: 'test1234',
+  text: 'test text',
+  object: {
+    testObject: 'data1234'
+  }
+}
+
+function setCustomData(data: any, name: string, format: string | undefined, isMultiple?: boolean) {
+  if (format === 'uri') data[name] = 'https://www.example.com'
+  if (name === 'email') data[name] = 'test@twilio.com'
+  if (name === 'email' && isMultiple) data[name] = ['test@twilio.com']
+  if (name === 'currency') data[name] = 'USD'
+  return data
+}
+
+export default function generateTestData(
+  destination: DestinationDefinition<any>,
+  action: ActionDefinition<any>,
+  flag: string
+) {
+  let eventData: any = {}
+  const settingsData: any = {}
+
+  const authentication = destination.authentication
+  if (authentication) {
+    for (const settingKey in authentication.fields) {
+      const type = authentication.fields[settingKey].type
+      settingsData[settingKey] = testData[type]
+      if (settingKey === 'endpoint') {
+        settingsData[settingKey] = 'https://www.example.com'
+      }
+    }
+  }
+
+  let fields = Object.keys(action.fields)
+
+  if (flag === 'required') {
+    fields = fields.filter((name) => action.fields[name].required || name.includes('id'))
+  }
+
+  for (const name of fields) {
+    const field = action.fields[name]
```

Comment:
```suggestion
  if (flag === 'required') {
    fields = fields.filter(() => action.fields[name].required || name.includes('id'))
  }

  for (const [name, field] of Object.entries(action.fields)) {
   if (flag === 'required' && (!field.required || !name.includes('id'))) {
     continue
   }
```

[`Object.entries`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries) is a helpful tool for iterating over objects when you need both the key and value